cmd/kube-proxy/app/server.go

```diff
@@ -287,6 +288,43 @@ func (o *Options) Validate(args []string) error {
 	return nil
 }
 
+// ValidateMixedArguments validates passed arguments
+func (o *Options) ValidateMixedArguments(flag *pflag.FlagSet) error {
+	// If --config isn't set, we have nothing to validate
+	if !flag.Changed("config") {
+		return nil
+	}
+
+	mixedInvalidFlags := []string{}
+	flag.Visit(func(f *pflag.Flag) {
+		if isAllowedFlag(f.Name) {
+			// allowed flags can be set with --config
+			return
+		}
+		mixedInvalidFlags = append(mixedInvalidFlags, f.Name)
+	})
+
+	if len(mixedInvalidFlags) != 0 {
+		return fmt.Errorf("can not mix '--config' with arguments %v", mixedInvalidFlags)
+	}
+	return nil
+}
+
+func isAllowedFlag(flagName string) bool {
+	knownFlags := sets.NewString("config",
```

Comment:
Would be better to define the list of allowed flags as constant / shared var with proper comment.